department-of-veterans-affairs/caseflow
# TYLERB/APPEALS-9138: Add Healthcare Provider option in dropdown in HLR flow

Resolves [APPEALS-9138](https://vajira.max.gov/browse/APPEALS-9138) and [APPEALS-9140](https://vajira.max.gov/browse/APPEALS-9140)

### Description
Added the ability to add unlisted Healthcare Provider Claimants during Higher-Level Review intake.

Still needs changes from two other stories to be completed.

### Acceptance Criteria
- [ ] The changes only affect HLR/SC claimant/poa forms and not appeals claimant/poa forms.
- [ ] The edit pages for these forms still work? SSN is up for grabs since I don't know how to switch it on/off for only HLR since that is a direct edit page.


### Testing Plan
1. Possibly data creation
2. Swap to an intake user.  BVAISHAW (VACO) is a common one.
3. Click on the mail intake tab and click the start link or navigate to the local url /intake (e.g. http://localhost:3000/intake/)
4. Select Decision Review Request: Higher-Level Review — VA Form 20-0996 from the intake dropdown to start a high level review intake
5. Click the continue to search button
6. Enter a valid veteran ssn into the input field and click the search button or hit enter to advance
7. Fill out all the fields the review request page.
8. Select 'Yes' for the field Is the claimant someone other than the Veteran?
9. Select Claimant not listed for the field Please select the claimant listed on the form. If the claimant is not listed, please select "Claimant not listed" and add their information in the next step.
10. Click the Continue to the next step button
11. Verify that the Relationship to the Veteran dropdown includes healthcare provider option
![image](https://user-images.githubusercontent.com/109369527/196780013-d551b64f-74be-4688-b359-502b2c5af8f3.png)

12. Select the healthcare provider option
13. Select organization for the radio button field Is the claimant an organization or individual?
14. Verify all the required fields and optional fields work according to the AC - lots of checks here
![image](https://user-images.githubusercontent.com/109369527/196781597-35b623e7-da03-4220-bbe1-cc15671868e6.png)

15. Select no for the radio button field Do you have a VA Form 21-22 for this claimant?
16. Select Continue to next step
17. Verify the information in the modal is correct based what you entered and the designs
![image](https://user-images.githubusercontent.com/109369527/196782059-a9e49885-3b2d-49b9-9246-24f781262eef.png)

18. Select Confirm on the modal
19. Verify that the claimant was added to the add_issues page
20. Do some code stuff to verify the data
21. Click cancel intake
![image](https://user-images.githubusercontent.com/109369527/196787290-bd623d7e-b676-4a44-98c9-49a247796e8e.png)

22. Cancel the intake and repeat steps 1-12
23. Select individual for the radio button field Is the claimant an organization or individual?
24. Verify all the required fields and optional fields work according to the AC - lots of checks here
25. Select no for the radio button field Do you have a VA Form 21-22 for this claimant?
26. Select Continue to next step
27. Verify the information in the modal is correct based what you entered and the designs
28. Select Confirm on the modal
29. Verify that the claimant was added to the add_issues page
30. Do some code stuff to verify the data
31. Finish out the intake and verify it succeeded 


### Storybook Story
*For Frontend (Presentational) Components*
- [ ] Verify changes are present in the AddClaimant storybook
- [ ] Verify changes are present in the Add PoA storybook

### Database Changes
*Only for Schema Changes*
- [ ] Verify that the ssn field has been added to the unrecognized party details table
